Example:

```go
var consumer *rabbitmq.Consumer
go func() {
    consumer = amqp.StartAmqpConsumer(conn, handler, key, queue)
}

// Some code here

consumer.Close()
```

Traceback:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0xf5048b2]

goroutine 9 [running]:
github.com/wagslane/go-rabbitmq.(*Consumer).Close(0x0?)
	/Users/user/go/pkg/mod/github.com/wagslane/go-rabbitmq@v0.14.2/consume.go:125 +0x12
```

It happens because consumer always equals to nil

--- a/amqp/amqp.go
+++ b/amqp/amqp.go
@@ -83,15 +83,16 @@ func NewAmqpConsumer(
 	conn *rabbitmq.Conn,
 	routingKey string,
 	queue string,
+	optionFuncs ...func(*rabbitmq.ConsumerOptions),
 ) (*rabbitmq.Consumer, error) {
-	return rabbitmq.NewConsumer(
-		conn,
-		queue,
+	defaultOptions := []func(*rabbitmq.ConsumerOptions){
 		rabbitmq.WithConsumerOptionsRoutingKey(routingKey),
 		rabbitmq.WithConsumerOptionsExchangeName(topic),
-		rabbitmq.WithConsumerOptionsConsumerAutoAck(true),
 		rabbitmq.WithConsumerOptionsLogger(CustomLogger{}),
-	)
+		rabbitmq.WithConsumerOptionsConsumerAutoAck(true),
+	}
+	optionFuncs = append(defaultOptions, optionFuncs...)
+	return rabbitmq.NewConsumer(conn, queue, optionFuncs...)
 }
 
 func StartAmqpPublisher(conn *rabbitmq.Conn) *rabbitmq.Publisher {

